File: src/main/java/com/example/qualitycontrolproject/entities/Project.java
package com.example.qualitycontrolproject.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long projectId;
    private String name;
    private String projectKey;
    private String type;
    private String url;
    @Column(name = "project_lead")
    private String lead;
    private String category;
    private String description;

    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Task> tasks;
}

File: src/web/src/app/classes/Project.ts
/*
export interface ProjectJiraDto {
  id: string;
  key: string;
  name: string;
  self: string;
  projectTypeKey: string;
  projectCategory?: {
    name: string;
    description: string;
  };
}

export interface ProjectCategory {
  name: string;
  description: string;
}
*/


export interface Project {
  projectId?: number;
  name: string;
  projectKey?: string;
  type?: string;
  url?: string;
  lead?: string;
  category?: string;
  description?: string;
  tasks?: any[];
}
